axiom/nr_analytics_events.c

```diff
@@ -117,11 +124,16 @@ int nr_analytics_events_number_saved(const nr_analytics_events_t* events) {
 #define NR_ANALYTICS_EVENTS_MAX_EVENTS_SANITY_CHECK (10 * 1000 * 1000)
 
 nr_analytics_events_t* nr_analytics_events_create(int max_events) {
-  nr_analytics_events_t* events;
-
   if (max_events <= 0) {
-    return 0;
+    return NULL;
   }
+
+  return nr_analytics_events_create_ex(max_events);
+}
+
+nr_analytics_events_t* nr_analytics_events_create_ex(int max_events) {
```

Comment:
1. Is there a reason not to make this function static and only accessible via nr_analytics_events_create? Is there a case where we want to access this directly without checking for NULL?
2. I would suggest a more descriptive name, like "_init" rather than "_ex", just for clarity-at-a-glance sake

Reply:
1. The `_ex` method is called by `nr_log_events_create`. That's why it was added in the first place - to allow log_events pool have the size of 0.
2. `_ex` is a standard suffix added when an interface of existing function is **__ex__**tendted.

Reply:
I missed the NRINLINE implementation by only searching .c files. Oops!

If we are going to allow the bypass of nr_analytics_events_create, do we need a check here for max_events < 0, or are we confident that will never happen?

Reply:
Good catch! I will add a check that `max_events` is not negative to `nr_log_events_create_ex`.

Reply:
@bduranleau-nr Addressed in 4e74eec94f98f5e1ed59c3f60eb92751c5411608.